ocaml/tools/ocamlprof.ml

```diff
@@ -358,24 +364,31 @@ and rewrite_ifbody iflag ghost sifbody =
     rewrite_exp iflag sifbody
 
 (* called only when !instr_fun *)
-and rewrite_annotate_exp_list l =
-  List.iter
-    (function
-     | {pc_rhs=Pboolean_guarded_rhs { pbg_guard; pbg_rhs }} ->
-         rewrite_exp true pbg_guard;
-         insert_profile rw_exp pbg_rhs;
-     | {pc_rhs=Ppattern_guarded_rhs { ppg_scrutinee; ppg_cases; _ }} ->
-         rewrite_exp true ppg_scrutinee;
-         rewrite_annotate_exp_list ppg_cases
-     | {pc_rhs=Psimple_rhs {pexp_desc = Pexp_constraint(sbody, _)}}
-       (* let f x : t = e *)
-       -> insert_profile rw_exp sbody
-     | {pc_rhs=Psimple_rhs sexp} -> insert_profile rw_exp sexp)
-    l
+and rewrite_annotate_exp_list l = List.iter rewrite_annotate_case l
+
+and rewrite_annotate_case case =
+  match Jane_syntax.Case.of_ast case with
+  | Some (jcase, _attrs) -> rewrite_annotate_case_jane_syntax jcase
+  | None ->
+  Option.iter (rewrite_exp true) case.pc_guard;
+  match case.pc_rhs.pexp_desc with
+  | Pexp_constraint (sbody, _) ->
+      (* let f x : t = e *)
+      insert_profile rw_exp sbody
+  | _ -> insert_profile rw_exp case.pc_rhs
```

Comment:
One goal of this PR is to reduce the diff with upstream &mdash; the total number of lines changed by this branch should go down. Can you spot-check the overall diff between the base of the `pattern-guards` branch and this commit to see if that in fact pans out?

Reply:
The number of changes actually looks pretty similar... do you think it's worth going through the diff with the branch base and manually reducing the number of changes when possible?